app/models/mission.rb

```diff
@@ -53,6 +53,53 @@ def java_main_contents
     end
   end
 
+  def java_pierced_contents
+    java_contents = java_contents_per_lines
+    pierced_locations.order('file_name DESC').order('location_id DESC').each do |p|
+      java_contents[p.file_name].slice!(p.lines[0], p.lines.length)
+    end
+    java_contents.map { |file, content| { file => content.join("\n") } }
+  end
+
+  def java_submitted_contents(submit_id)
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Metrics/AbcSize: Assignment Branch Condition size for java_submitted_contents is too high. [21.02/15]